### ISSUE TITLE
Documentation for libmgba.

### ISSUE THREAD

**Issue by sergiotarxz** [mgba-emu/mgba#2847]
Hi, mgba developers.

I am currently making use of the libmgba library for my own client-server experiments with multiplayer and after a lot of time reading src/platform/qt files I am starting to grasp how to work with libmgba.

I really think this is a great library plus a great program to emulate gba, but documentation and examples do not exists anywhere you can search for them. (If you do not count code itself.)

I would like to contribute a doxygen starting point so relevant for the library subroutines, structs and enums can be consulted with less mental overhead in a html endpoint, maybe somewhere in mgba.io or maybe in the github web page or at least future developers will be able to compile the docs with the project to get something to stick with.

At first such documentation will be incomplete, but I think the doxygen option to show every symbol no matter if it is documented or not can really help those starting their way with libmgba meanwhile the most relevant parts of the API are documented.

I am opening this issue to hear your opinions about how this should be done if it should be done at all.

Regards, Sergiotarxz.

**Comment by endrift** [comment]
Adding documentation for libmgba [is definitely in the plans](https://mgba.io/2022/10/11/mgba-0.10.0/#what-about-10), but given the current flux of the API, I haven't yet committed to starting on it--especially since nothing in internal/ should really be used, as it's not considered stable or meant to be exposed. I'm also on the fence about what documentation system to use. doxygen is, well, clunky. I've used it before and it often is kind of bad. I'm not too sure about the alternatives yet, considering I wasn't planning to really start this until 0.12 or so.

I should also note that there is a bit of [heavily-commented example code](https://github.com/mgba-emu/mgba/tree/master/src/platform/example/client-server), but it's pretty minimal at the moment.

**Comment by sergiotarxz** [comment]
The client-server example is useful, but it does not use for example mCoreThread which to my understanding is a prerequisite to really achieve multiplaying in any client and also does a big bunch of work for you when you have to setup multiple instances.

I am happy to hear that documentation is on the plans, I did not read that release message.

**Comment by endrift** [comment]
Yeah, it's mentioned in a few places, but nowhere particularly obvious.

**Comment by endrift** [comment]
If major things like meson and gstreamer use it, despite Debian or Gentoo not shipping it, I wouldn't be too worried tbh.
